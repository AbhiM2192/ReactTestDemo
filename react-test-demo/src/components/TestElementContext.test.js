import React from 'react';
import {render,cleanup,fireEvent} from '@testing-library/react';

import NameProvider,{TestElementContext,NameContext} from './TestElementContext';

const renderWithContext = (component) =>{
    return {
        ...render(
            <NameProvider value={NameContext}>
                {component}
            </NameProvider>
        )
    } 
}

afterEach(cleanup)

it('Should Capture snapshot',() =>{
    const {asFragment} = renderWithContext(<TestElementContext/>)
    expect(asFragment(<TestElementContext/>)).toMatchSnapshot();
});

it('Should contain empty text onLoad',() =>{
    const{getByTestId} = renderWithContext(<TestElementContext/>)
    expect(getByTestId('Name')).toHaveTextContent('');
})

it('Should change text on click',() =>{
    const {getByTestId} = renderWithContext(<TestElementContext/>);
    fireEvent.click(getByTestId('changeName'));
    expect(getByTestId('Name')).toHaveTextContent('React Context test')
})