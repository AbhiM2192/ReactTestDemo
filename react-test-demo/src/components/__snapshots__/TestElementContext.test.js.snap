// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Capture snapshot 1`] = `
<DocumentFragment>
  <h1
    data-testid="Name"
  />
  <button
    data-testid="changeName"
  >
    Change Name
  </button>
</DocumentFragment>
`;
